### ISSUE TITLE
[Feature request] Possibe refinement of tag summary function

### ISSUE THREAD

**Issue by Ravenclawer** [moremeyou/Obsidian-Tag-Buddy#12]
When using the below syntax to summarise certainly tags, the images in the insulting view are plain texts instead of rendered images.
```tag summary
include: #tag1 #tag2
```
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/e066160c-9c51-4905-a41e-80668c7241c1)

# Currently workaround  
Using the flatten summary button to transport all the search results to the current note, and images will be rendered as they are embedded in the current note.

# Problems with a current workaround
One of the possible problems with using the flatten function to summarise headings is that may cause multiple versions of certain headings since one tends to edit the summarised results. Such editing cannot synced to the original heading.
 Another problem is that the flattened results will be displayed in the future tag summary syntax causing clutter and might be confusing.

# Possible solution
1. Make images rendered in the tag summary syntax.
2. (might be harder but more practical) Making an option to embed the heading of the tag summary results in the current note. (I often add tags to the headings of notes, for tags added at the end of a paragraph, maybe embedding the closest heading above the tags could be a possible solution)

**Comment by moremeyou** [comment]
Should be fixed now. It's the same versions number, so you'll need to uninstall and reinstall the plugin.

**Comment by Ravenclawer** [comment]
Thank you for the fix. The images is rendered now. is it possible for you to implement "Making an option to embed the heading of the tag summary results in the current note. (I often add tags to the headings of notes, for tags added at the end of a paragraph, maybe embedding the closest heading above the tags could be a possible solution)" if you have time?

**Comment by Ravenclawer** [comment]
Currently, the flatten summary button can copy the summary contents to the current note, but this leaves possible risks of causing duplication in the future summary and version differences between the same content

**Comment by moremeyou** [comment]
Hey there. I'm curious to get a better understanding of the use case. What do you mean by the heading of the tag summary - something outside the summary code block? For your second point, the plugin will exclude any queried tags in the note where the summary is generated.

**Comment by Ravenclawer** [comment]
When I read research papers, I often collect all the images of the relevant papers in a single note. Every image has its headings and I can tag the image in its respective heading with the information I'd like to extract from the image. The heading contains a double bracket link to their citekey (Zotero generated) which could help me to identify the source of the image. the following image is an excerpt of such a note.
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/a107afad-e63d-47d2-a5a7-7e75e9c37c79)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/19a75f8f-2b0f-485f-8b5e-3265ca287a92)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/7540c243-2e4a-44bb-89fc-c158b215679d)
Since tag buddy allows tags at the end of the paragraph, some of my old notes have such paragraphs that have tags at the end. So I mimic the scenario in the above example. 

The above note could be considered my knowledge base
Let's say one day, my professor finds me and tells me that, I found microalgae could be a good research topic, what do you think, can you tell me what resource you need if you have to carry out related research?

Then I will create a new note and conduct a survey about microalgae.
I will have a tag summary query about microalgae first
```tag-summary
include: #microalgae
```
Then it will help me find all the tagged headings and paragraphs 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/dd1f33bf-0e1c-45d0-b530-0698273c6036)

Now let's say we are particularly interested in rapid curve and want to survey how it could be measured or produced 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/a2ce03e8-1b77-4cce-aa85-f022d5c6c2c1)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/285b09bf-8b69-4dc6-8753-17a87b7adadb)
If I use ctrl + hover on the link, I can open the note and edit it, but I have to scroll down to find the tagged heading. 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/ec0dbd2b-18fb-43d7-9835-f7ff6800fbc8)
If I use the flatten summary button, I can further edit the rapidcurve part but the note will not sync to the original heading, if I query something in a new note like test3, there will be two results with overlapped information, which is confusing when such notes added up.
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/552bd1a4-9f6a-4fdd-a079-00a215943a4a)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/dbfb8942-c608-4f7a-a1b9-b65c8fa383e0)

Using Heading embed could edit the content in the hover editor and will not cause duplication in future search 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/b280dd32-8165-49c2-9c3e-42f3d8178193)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/5de859ed-a4b3-499b-81b4-2bb0c583c1f7)
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/82498ab2-3032-4cb2-8d1c-e36a841aac7a)

**Comment by Ravenclawer** [comment]
This method allows incremental reading of research papers, one can query any tagged contents and further edit it without worrying about causing duplication or version differences. I think it's a great feature.

**Comment by moremeyou** [comment]
Wow, super thorough. And I feel you with the need to make it work in just the way you want. It's why I made the plugin, nothing working quite I like I wanted. That said, I *think* I understand the functionality now. Unfortunately, tagged content is grouped by "blocks." That is, any block of text between empty lines. The 4 lines below are one block.

something #tag
## Some heading #anotherTag
Details
This and that.

So that empty space knowledge might be the detail you need to make it work how you want. Otherwise, detecting/pulling a heading from one of the summary items is out of scope right now. However, at one point I generated a heading when baking the summary. I've since moved that to when you flatten the summary to a new note. I could bring that kind of thing back as a setting for baking within the note. But check it out, see if it would work for your needs.

I'm also curious if you've used the dateview/query plugin? It's much more robust and might be better suited.

**Comment by Ravenclawer** [comment]
I see, you mean that Obsidian offers no function to locate headings with tags, and when a tag is found in a note, the whole page will be considered a block, and it's impossible to apply heading embed or heading link in the plugin.

**Comment by Ravenclawer** [comment]
I reinstall the plugin and haven't see any thing like this
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/a3d6d467-90b1-4910-b306-2ccfce04582b)

**Comment by Ravenclawer** [comment]
Thank you for your time, I will try dataview and see if there is a solution to this

**Comment by Ravenclawer** [comment]
Could you make this link a link to heading instead of note？
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/c5a47a69-7eef-4099-936c-d642995b620c)

**Comment by Ravenclawer** [comment]
The closest query for this that I found could not address the issue, it could only query the headings with fixed name https://share.note.sx/47oiqm8g

**Comment by moremeyou** [comment]
I assume test_1 is the note link generated for each of the summary items? If so, I think that could be possible. So instead of it linking to the source note, it would look for the first heading in the block and make an anchor link to that. How does that sound?

About the missing buttons, check setting and switch the visibility options in case the old settings haven't updated. Also, might need to completely uninstall the plugin and reinstall. Only need to do this while I make these minor tweaks with the same version (not best practice, I know :)

I mention testing the buttons because copying a summary item to a section/heading in the current note (or another note) it _does_ anchor link to the heading specified (if it exists).

**Comment by Ravenclawer** [comment]
Thank you for putting your valuable time into this feature. I completely reinstalled the plugin and found the button was refreshed, but the function changed, and some of them not working properly
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/8c317074-3dad-4547-834b-f6702667f318)

The copy paragraph link function produce link to the note 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/69707e7d-4483-4103-bb68-589b702e0f68)

"I assume test_1 is the note link generated for each of the summary items? If so, I think that could be possible. So instead of it simply linking to the top of the source note, it would look for the first heading in the block of that summary item and make an anchor link to that. How does that sound?"
this is exactly what I want

**Comment by Ravenclawer** [comment]
> Could you make this link a link to heading instead of note？ ![image](https://private-user-images.githubusercontent.com/73892411/319824134-c5a47a69-7eef-4099-936c-d642995b620c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIzNjY2MDEsIm5iZiI6MTcxMjM2NjMwMSwicGF0aCI6Ii83Mzg5MjQxMS8zMTk4MjQxMzQtYzVhNDdhNjktN2VlZi00MDk5LTkzNmMtZDY0Mjk5NWI2MjBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA2VDAxMTgyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlN2E0N2I3YjQxYTcyZDQwNTQzNTQ4Yzc3MzkwYTlmMzY2OTBjMGMyMWE2OTE5MWUwNmI1ZDc0ZGY1OGFhMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uMmhdYIap04hwd_B5FSx-BfgaLhDAfhayrv2oNy8RZ8)

[[test1]] in the tag buddy summary code blocks
could be 
[[test1#heading（where tags were found）]]

**Comment by Ravenclawer** [comment]
The previous example might be to long to demonstrate my use case 
here is a more concise one:
# note 1
---
title: test 1
contents:
# heading 1
paragraph1 
#tag1 

# #tag2 heading 2
paragraph2

# #tag1 heading 3
paragraph3
---
# note2
---
title: test2
```tag-summary 
include: #tag1
```
---

# The query result could be 
[[test1#heading 1]]
<tag buddy block for the content of test 1#heading1>
[[test1#tag1 heading 3]]
<tag buddy block for the content of test1#tag1 heading 3>
(note: the # mark will be ignored by the obsidian heading link)
OR
![[test1#heading 1]]
![[test1#tag1 heading 3]]
(note: the # mark will be ignored by the obsidian heading link)

Instead of
[[test1]]
<tag buddy block for the content of test 1#heading1>
[[test1]]
<tag buddy block for the content of test1#tag1 heading 3>
(note: the # mark will be ignored by the obsidian heading link)

**Comment by Ravenclawer** [comment]
[test 1.md](https://github.com/moremeyou/Obsidian-Tag-Buddy/files/14891961/test.1.md)
[test 2.md](https://github.com/moremeyou/Obsidian-Tag-Buddy/files/14891962/test.2.md)
[test 3.md](https://github.com/moremeyou/Obsidian-Tag-Buddy/files/14891963/test.3.md)

**Comment by moremeyou** [comment]
For the anchor linking. Should be no problem. I'll put it on the todo list. A lil donation would be appreciated tho :)
When TB finds a tagged block, it will look for the first instance of a heading and used that for the link.

> completely reinstalled the plugin and found the button was refreshed, but the function changed, and some of them not working properly

I'd need to investigate the file it's trying to copy to. Can you send it, or let me know if there's any special kind of formatting at the end of the file?

**Comment by Ravenclawer** [comment]
Maybe I misunderstood the button‘s function, I pressed the copy paragraph link button in a new note named untitled 1 and found the resulting link was a link to note the note title instead of the paragraph. maybe the formatting of the test1.md caused the malfunction, Which I have already uploaded.
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/9fb53b23-17fd-43d1-bf98-c0dbcd32fa52)

As for donations, I want to support the development of the plugin. However, I cannot access a visa or any international payment system. I have access to Alipay, which is not supported by the website.

**Comment by Ravenclawer** [comment]
There are two options in the 0.6.8 version of the plugin 
![image](https://github.com/moremeyou/Obsidian-Tag-Buddy/assets/73892411/925057b0-d65c-431c-82e9-9d396b1ffaf2)

**Comment by Ravenclawer** [comment]
Hi. Although I can't donate for the time being. I promise will write a showcasing article in the community forum to share my workflow and help promote the plugin, a game changer that allows using tags in Obsidian to make a section-level search and collection of information, which I have never seen before (Tag summary only allow paragraph level of query and  collection of information)

**Comment by Ravenclawer** [comment]
Hi. Although I can't donate for the time being. I promise will write a showcasing article in the community forum to share my workflow and help promote the plugin, a game changer that allows using tags in Obsidian to make a section-level search and collection of information, which I have never seen before (Tag summary only allow paragraph level of query and  collection of information)

**Comment by moremeyou** [comment]
That'd be great. Jammed up with work right now. But I'll check back when I
get into another sprint :)

**Comment by moremeyou** [comment]
Hey @Ravenclawer,
I've implemented this feature. It will be released with the next update. But if you want to use it right now, download the [latest main file](https://github.com/moremeyou/Obsidian-Tag-Buddy/blob/e35552ac2a06f713d967c09a825e405b2f9aec30/src/main.ts) and replaces it in the Tag Buddy plugin folder.

Note that the link NAME will still be the file name. But the link itself will go to the closest header in that tagged paragraph. I'd love to hear what you think and/or if you have any issues.

D

**Comment by Ravenclawer** [comment]
Thank you for implementing the function, maybe I have used the wrong file, I have download the main.ts file and changed the file name to main.js. Then, I replaced the file in the obsidian plugin folder. When I came back to obsidian community plugin interface, the Tag buddy cannot be activated.

**Comment by moremeyou** [comment]
Hmm, I had to upload the dev environment recently. Maybe it needs
supporting files on my end. Try this pre-made .js file
https://github.com/moremeyou/Obsidian-Tag-Buddy/blob/main/src/main.js

D

**Comment by Ravenclawer** [comment]
Thank you for making a pre-release, currently the generated syntax cannot be rendered and jump to the paragraph. I think it's the addition of tags in the heading that caused the problem. Please check out the attached md file, and put them into an ob
[test1-database.md](https://github.com/user-attachments/files/18038332/test1-database.md)
[test2-lookup.md](https://github.com/user-attachments/files/18038335/test2-lookup.md)
sidian vault and you can better understand the problem。

**Comment by Ravenclawer** [comment]
Maybe obsidian developed a strategy to omit some special letters in the heading to avoid confusion

**Comment by moremeyou** [comment]
Hey, thanks for sharing those files.
I think I've cracked it. Try this main.js file
<https://github.com/moremeyou/Obsidian-Tag-Buddy/blob/main/src/main.js>.
I think I've reached the limit of how many tweaks I can make to the link
generation. Tbh, I'm surprised these complex links work. Generally I'd
suggest clean headers. Keep any complex formatting out of headers you want
to link to. But I am pleased it seems to be working.

D
